source/cloud/terraform.md

```diff
@@ -286,40 +215,104 @@ you a message after creating all the required resources.
-Creation complete after 1m35s [id=88c7d037-5c44-45b7-acce-f5e4e58b1c35]
-Apply complete! Resources: 4 added, 0 changed, 0 destroyed.
+Apply complete! Resources: 6 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+MyVM = "SSH: ssh -A -p 56315 rocky@193.190.80.3"
+```
+
+:::{tip}
+Usually the username (rocky/ubuntu/...) will be provided by `terraform output`. If you forget or are not sure what username to use, just connect with `root` and a message will appear telling you the correct user.
+:::
+For a cluster it might look like this:
 ```
+..
+..
+
+Outputs:
+
+cluster = <<EOT
+Main:
+SSH: ssh -A -p 59619 rocky@193.190.80.3
 
+Private:
+Note: SSH to main server first
+MyCluster-private-0:
+ SSH: ssh -A rocky@10.113.1.135
+MyCluster-private-1:
+ SSH: ssh -A rocky@10.113.1.229
+MyCluster-private-2:
+ SSH: ssh -A rocky@10.113.1.213
+
+EOT
+```
+:::{tip}
+To access your private VMs you need to ssh to your public VM first, then ssh from your main vm to your private VMs.
+:::
 Your cloud infrastructure is ready to be used.
 
+:::{tip}
+If you forgot your VM's details, just run `terraform output`
+If you make any changes to the template, just run `terraform apply` again.
+:::
+
 :::{tip}
 It is important to keep a backup of your terraform directory, specially
 all the files within the environment directory:
 `~/openstack-templates/terraform/environment`
 :::
 
 
+
 Terraform generates several files in this directory to keep track of any
 change in your infrastructure. If for some reason you lost or remove
 these files you will not able to modify or change the current Terraform
 plan (only directly from OpenStack).
 
 
 :::{warning}
-It is possible to execute `./modify_variable.sh` script several times,
-(for instance, if you have changed the variables within
-`modify_variable.config` file). This will update your current Terraform
-deployment. Any Terraform variables change may affect to your current
-running VM(s) as well, `terrafrom apply` will inform you about this and
-it will request confirmation. The external ports for port forwarding
-are generated randomly when you run `modify_variable.sh` by default, but
-you can keep the current port forwarding rules if you add `-k` option:
-`./modify_variable.sh -k`
+Do not remove or modify the port_something_ssh.json files. These keep track of the external ports used. Deleting them _will_ break Terraform.
+:::
+## Making changes to module variables
+### Adding NFS Or Nginx
+:::{tip}
+If you enabled these options upon initial creation of the VM, you don't have to do these steps
 :::
 
+If you added NFS or Nginx after initial creation, you need to run a script **on your openstack VM** to mount the NFS volume or install nginx, respectively.
+
+First ssh to your instance. You can get the command with
+```shell
+terraform output
+```
+Then run the command you got from terraform.
+After connecting to your instance, run:
+```shell
+curl http://169.254.169.254/2009-04-04/user-data | sudo bash
```

Comment:
this is an example or always this command?

Reply:
Surprisingly yeah its always this command